docs/50_Writable_Views.md

```diff
@@ -0,0 +1,521 @@
+         "preserveNullAndEmptyArrays": true
+      }
+   },
+   {
+      "$project": {
+         "riderId": 0
+      }
+   }
+]
+```
+
+resulting in the following aggregated record:
+
+```json title=Order Details
+{
+   "_id": "6554e9eedf0efad9177c8212",
+    "date" : "2023-11-17T14:17:08.770Z",
+    "totalAmount" : 12.5,
+    "items" : [
+      // ...
+    ],
+    "rider" : {
+        "firstName" : "Frodo",
+        "lastName" : "Baggins",
+        "ordersDelivered" : 1.0,
+        "tipPercent" : 13.0,
+        "transportationMean" : "eagle"
+    }
+}
+```
+
+This looks great in case your application just need to visualize those data aggregated. However,
+it poses a limitation whenever it is necessary to add, edit or delete references between the two collections. In fact, applications using CRUD Service would require to call the source collection endpoint, which adopts a different data model, leading to duplicate the logic to read and change object references.
+
+---
+
+## Overview
+
+To address the problem mentioned in the previous section, CRUD Service provides a feature called _writable views_. This type of views are **limited** on what can actually be written back to the source collection and they require the adoption a specific format for lookup fields. However, the benefit is that they allow to **change lookup references** on the source collection using the same interface of the view endpoint.
+
+_Writable views_ feature can be enabled per CRUD Service view by setting the property `enableLookup` to `true` in the view definition model. This allows the service to create expose further `POST`, `PATCH`, and `DELETE` routes in addition to a view existing `GET` endpoints. These new routes gives the caller the possibility to add, change or delete records on the source collection using the view data model. In particular:
+
+- `POST`: allow adding a new record on the source collection
+- `PATCH`: allows editing the lookup fields on the source collection
+- `DELETE`: allow removing an existing record from the source collection
+
+Please note that there are a few **limitations**, that will be briefly discussed.
+
+Moreover, CRUD service exposes an additional route for each `$lookup` step found in the **first-level** of view's aggregation pipeline. Accessing these routes, denoted as `/lookup/<name of the view field>`, provides access the records coming from the _lookup_ collection, so that it is possible to list all the values (`ObjectId`) that can be assigned to that particular lookup field.  
+Please note that resulting records are returned adopting the same format specified in the lookup pipeline.
+
+## Limitations
+
+Before enabling this feature on a CRUD Service view it important to consider the following limitations:
+
+- view aggregation pipeline must have at least one `$lookup` step in the view pipeline. On the contrary the feature would be useful, since there would not be any reference to other collection to be edited;  
```

Comment:
```suggestion
- view aggregation pipeline must have at least one `$lookup` step in the view pipeline. On the contrary the feature would not be useful, since there would not be any reference to other collection to be edited;  
```